### ISSUE TITLE
modal.update() supports functional updating

### ISSUE THREAD

**Issue by Mongkii** [ant-design/ant-design#27162]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
The current `modal.update()` only allows the `newConfig` object as argument, and uses shallow merge (`{ ...oldConfig, ...newConfig }`) to generate new config.

Therefore when I want to update a value in a nest object of `config` (like updating part of properties in `config.okButtonProps`), it can cause inconvience.

That's what happened in the Story below. You may view the Story in CodeSandbox.
[CodeSandbox](https://codesandbox.io/s/friendly-breeze-mkthf?file=/src/App.tsx)

If the `modal.update()` supports functional updating ( `(prevConfig) => newConfig` ), it would be much easier to do a modal update.

### What does the proposed API look like?
```typescript
type ConfigUpdate = ModalFuncProps | ((prevConfig: ModalFuncProps) => ModalFuncProps);

function update(configUpdate: ConfigUpdate): void;
```

**Comment by Mongkii** [comment]
I've created a pull request to add this feature: #27163

**Comment by Mongkii** [comment]
PR merged in v4.8.0
